### PR TITLE
CY-933 Fix input validations

### PR DESCRIPTION
**Pull request by davidginzbourg**
- Fixes regex string validation: it now uses basestring and not str.
- Fixes basic data types not validated during deployment creation.

### CODE REVIEW

File: dsl_parser/tests/test_inputs.py [hunk 1]
```diff
@@ -1152,80 +1154,121 @@ def test_input_validate_type_raises(self):
...
+
+    def test_input_validate_boolean_raises_with_deployment(self):
+        self._test_input_validate_type_raises_with_deployment('boolean', 'x')
+
+    def test_input_validate_string_successful(self):
+        self._test_input_validate_type_successful('string', 'some_string',
+                                                  self.assertEqual)
+        # Using type: string, allows us to pass any kind of type (due to
+        # backward compatibility)
+        self._test_input_validate_type_successful('string', list(),
```
Comment (tyacbovi) [inline dsl_parser/tests/test_inputs.py]: why does "list()" works as a type name?

File: dsl_parser/tests/test_inputs.py [hunk 2]
```diff
@@ -1152,80 +1154,121 @@ def test_input_validate_type_raises(self):
...
+        self._test_input_validate_type_raises_with_deployment('float', True)
+
+    def test_input_validate_boolean_successful(self):
+        self._test_input_validate_type_successful('boolean', False,
+                                                  self.assertEqual)
+
+    def test_input_validate_boolean_raises(self):
+        self._test_input_validate_type_raises('boolean', [])
+
+    def test_input_validate_boolean_raises_with_deployment(self):
```
Comment (tyacbovi) [inline dsl_parser/tests/test_inputs.py]: improve tests names, they are not reflecting the cause that the test failed

File: dsl_parser/tests/test_inputs.py [hunk 3]
```diff
@@ -1152,80 +1154,121 @@ def test_input_validate_type_raises(self):
...
+    def test_input_validate_integer_raises_with_deployment(self):
+        self._test_input_validate_type_raises_with_deployment('integer', 'x')
+        self._test_input_validate_type_raises_with_deployment('integer', True)
+
+    def test_input_validate_float_successful(self):
+        self._test_input_validate_type_successful('float', 123,
+                                                  self.assertEqual)
+        self._test_input_validate_type_successful('float', 123.1,
+                                                  self.assertEqual)
+        self._test_input_validate_type_successful('float', long(),
```
Comment (tyacbovi) [inline dsl_parser/tests/test_inputs.py]: the other way will work? because u lose accuracy

File: dsl_parser/tests/test_inputs.py [hunk 4]
```diff
@@ -1152,80 +1154,121 @@ def test_input_validate_type_raises(self):
...
+        self._test_input_validate_type_raises('boolean', [])
+
+    def test_input_validate_boolean_raises_with_deployment(self):
+        self._test_input_validate_type_raises_with_deployment('boolean', 'x')
+
+    def test_input_validate_string_successful(self):
+        self._test_input_validate_type_successful('string', 'some_string',
+                                                  self.assertEqual)
+        # Using type: string, allows us to pass any kind of type (due to
+        # backward compatibility)
```
Comment (tyacbovi) [inline dsl_parser/tests/test_inputs.py]: backward compatibility of a Cloudify feature?

File: dsl_parser/tests/test_inputs.py (hunk 1 above)
Comment (davidginzbourg) [reply dsl_parser/tests/test_inputs.py]: It's the the type name, it's the value, i.e. an empty list.

**Pull request merged**